Source/WebCore/platform/gtk/ScrollbarThemeGtk.cpp

```diff
@@ -46,19 +45,19 @@ ScrollbarTheme& ScrollbarTheme::nativeTheme()
     return theme;
 }
 
-ScrollbarThemeGtk::~ScrollbarThemeGtk() = default;
-
-static void themeChangedCallback()
+ScrollbarThemeGtk::~ScrollbarThemeGtk()
 {
-    ScrollbarTheme::theme().themeChanged();
+    SystemSettingsGLib::singleton().removeObserver(this);
 }
 
 ScrollbarThemeGtk::ScrollbarThemeGtk()
 {
     static bool themeMonitorInitialized = false;
     if (!themeMonitorInitialized) {
-        g_signal_connect(gtk_settings_get_default(), "notify::gtk-theme-name", G_CALLBACK(themeChangedCallback), nullptr);
-        g_signal_connect(gtk_settings_get_default(), "notify::gtk-overlay-scrolling", G_CALLBACK(themeChangedCallback), nullptr);
+        SystemSettingsGLib::singleton().addObserver([](const SettingsStateGLib& state) mutable {
+            if (state.themeName || state.overlayScrolling)
+                ScrollbarTheme::theme().themeChanged();
+        }, this);
```

Comment:
Since this is a singleton, I guess this could be notified from the SystemSettings like we do with RenderTheme